pulp_container/app/viewsets.py

```diff
@@ -946,8 +949,13 @@ def build_image(self, request, pk):
             containerfile.touch()
         tag = serializer.validated_data["tag"]
 
-        artifacts = serializer.validated_data["artifacts"]
-        Artifact.objects.filter(pk__in=artifacts.keys()).touch()
+        artifacts, repo_version = None, None
+        if serializer.validated_data.get("artifacts"):
+            artifacts = serializer.validated_data["artifacts"]
+            Artifact.objects.filter(pk__in=artifacts.keys()).touch()
+        elif serializer.validated_data.get("repo_version"):
+            repo_version = serializer.validated_data["repo_version"]
+            self._check_file_repo_permission(request, repo_version)
```

Comment:
Instead of doing this, we may want to declare a new entry inside the access policy, like we did for the repository viewset:
https://github.com/pulp/pulp_container/blob/25abfd6cb0bff1ac8756e4d15c65b83839594f66/pulp_container/app/viewsets.py#L674-L682